File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/3042.py
while True:
    m = int(input())
    if m == 0: break
    
    pos = 1
    mov = 0
    
    for i in range(m):
        menor = 3
        k = [int(j) for j in input().split()]
        
        for j in range(0, 3):
            linha = k[j]
            if(linha == 0 and abs(j-pos) < menor):
                menor = abs(j-pos)
                posd = j
        pos = posd
        mov += menor
    print(mov)
        

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/3042.c
#include <stdio.h>

int abs(int n){
    if(n < 0) return n * (-1);
    return n;
}

int main(){
    while(1){
        int i, j, m, pos=1, mov=0, linha, menor, posd;
        scanf("%d", &m);
        if(m==0) break;
        for(i=0;i<m;i++){
            menor = 3;
            for(j=0;j<3;j++){
                scanf("%d", &linha);
                if(linha==0 && abs(j-pos)<menor){
                    menor = abs(j-pos);
                    posd = j;
                }
            }
            pos = posd;
            mov+=menor;
        }
        printf("%d\n", mov);
    }
 
    return 0;
}